managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -222,7 +222,9 @@ protected void recoverFromLedger(final ManagedCursorInfo info, final VoidCallbac
         // a new ledger and write the position into it
         ledger.mbean.startCursorLedgerOpenOp();
         long ledgerId = info.getCursorsLedgerId();
+        final long now = System.currentTimeMillis();
```

Comment:
Always use `System.nanoTime()` for measurements because it guarantees monotonically increasing timestamp. `currentTimeMillis()` can go backward (NTP syncup, Daylight saving time adjustments, etc.)